libs/RobotKit/tests/RobotController_test_stateWorking.cpp

```diff
@@ -79,16 +79,54 @@ TEST_F(RobotControllerTest, stateWorkingEventEmergencyStop)
 	EXPECT_TRUE(rc.state_machine.is(lksm::state::emergency_stopped));
 }
 
-TEST_F(RobotControllerTest, stateWorkingEventAutonomousActivityRequested)
+TEST_F(RobotControllerTest, stateWorkingDiceRollDetectedDelayNotOver)
 {
 	rc.state_machine.set_current_states(lksm::state::working);
 
+	auto maximal_delay_before_over = 1s;
+
+	EXPECT_CALL(mock_videokit, displayImage).Times(0);
+
+	spy_kernel_addElapsedTimeToTickCount(maximal_delay_before_over);
+	rc.onMagicCardAvailable(MagicCard::dice_roll);
+
+	EXPECT_TRUE(rc.state_machine.is(lksm::state::working));
+}
+
+// ! TODO: Refactor with composite SM & CoreTimer mock
+TEST_F(RobotControllerTest, stateWorkingDiceRollDetectedDelayOverEventAutonomousActivityRequested)
+{
+	rc.state_machine.set_current_states(lksm::state::working);
+
+	auto minimal_delay_over = 1001ms;
+
 	Sequence on_exit_working_sequence;
 	EXPECT_CALL(timeout, stop).InSequence(on_exit_working_sequence);
 
 	EXPECT_CALL(mock_videokit, displayImage).Times(1);
 
+	spy_kernel_addElapsedTimeToTickCount(minimal_delay_over);
 	rc.onMagicCardAvailable(MagicCard::dice_roll);
 
 	EXPECT_TRUE(rc.state_machine.is(lksm::state::autonomous_activities));
 }
+
+// ! TODO: Refactor with composite SM & CoreTimer mock
+TEST_F(RobotControllerTest, stateWorkingImpossibleSituationActivityStarted)
+{
+	rc.state_machine.set_current_states(lksm::state::working);
+
+	auto maximal_delay_before_over = 1s;
+
+	const std::unordered_map<MagicCard, interface::Activity *> activities {{MagicCard::number_10, &display_tag}};
+	activitykit.registerActivities(activities);
+
+	EXPECT_CALL(mock_videokit, displayImage).Times(1);
+
+	activitykit.start(MagicCard::number_10);
```

Comment:
You can refactor this one as you did for `stateGameActivityStartedBackToMenu`

```suggestion
	auto set_activitykit_is_playing = [this] {
		const std::unordered_map<MagicCard, interface::Activity *> activities {{MagicCard::number_10, &display_tag}};
		activitykit.registerActivities(activities);
		EXPECT_CALL(mock_videokit, displayImage).Times(1);
		activitykit.start(MagicCard::number_10);
	};
	set_activitykit_is_playing();
```